libgnucash/app-utils/options.scm

```diff
@@ -470,6 +470,13 @@ the option '~a'."))
 	 (selection-budget initial-budget)
```

Comment:
`gnc_budget_get_default` returns a `GncBudget*` so ISTM you need to change this to
```
(selection-budget (convert-to-guid initial-budget))
```